### PR TITLE
SDL watchdog

### PR DESCRIPTION
**Pull request by LuxoftAKutsan**
Create service responsible for running and monitoring multiple SDL processes and handling SDL configurations.
Service will provide API to test scripts, and script will be able to run SDL with certain configuration. 
Also this service should provide possibility to run multiple SDL instances (for parallel testing), monitor and control their life cycle.

### CODE REVIEW

**Comment by AStasiuk** [comment]
@theresalech proposal is ready for review

File: proposals/nnn-atf-sdl-watchdog-service.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+Create service responsible for running and monitoring multiple SDL processes and handling SDL configurations.
+Service will provide API to test scripts, and script will be able to run SDL with certain configuration. 
+Also this service should provide possibility to run multiple SDL instances (for parallel testing), monitor and control their life cycle. 
+
+## Motivation
+
+Major problem of current automated testing approach is that it is impossible to test multiple instances of SDL simultaneously.
+Currently SDL and ATF should be located on the same workstation because ATF actually runs SDL via bash scripts. 
+Implementation of SDL watchdog gives possibility to locate ATF and SDL independently (on one or on separate workstations).
+Also ATF is not able to control SDL configuration, it just uses SDL that located in predefined place.
```
Comment (theresalech) [inline proposals/nnn-atf-sdl-watchdog-service.md]: Please update to "...it just uses SDL that is located in a predefined place."

File: proposals/nnn-atf-sdl-watchdog-service.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+Service will provide API to test scripts, and script will be able to run SDL with certain configuration. 
+Also this service should provide possibility to run multiple SDL instances (for parallel testing), monitor and control their life cycle. 
+
+## Motivation
+
+Major problem of current automated testing approach is that it is impossible to test multiple instances of SDL simultaneously.
+Currently SDL and ATF should be located on the same workstation because ATF actually runs SDL via bash scripts. 
+Implementation of SDL watchdog gives possibility to locate ATF and SDL independently (on one or on separate workstations).
+Also ATF is not able to control SDL configuration, it just uses SDL that located in predefined place. 
+If script be able to get SDL with certain build flags, or certain version of SDL, that will reduce a lot of testing efforts.
```
Comment (theresalech) [inline proposals/nnn-atf-sdl-watchdog-service.md]: Please update to "If script is able to get..."

File: proposals/nnn-atf-sdl-watchdog-service.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+ - Callback triggered by watchdog service when SDL stops
+Options:
+ - ID of SDL process
+ - Stop reason (crash, regular shutdown, kill, etc) 
+ - Payload (core file in case if SDL crashed) 
+```
+
+#### ATF internal implementation
+
+On ATF side should be created component (for example: `SDL`) that is able to communicate with SDL Watchdog via API.
```
Comment (theresalech) [inline proposals/nnn-atf-sdl-watchdog-service.md]: Please update to "On ATF side, should create a component..."

File: proposals/nnn-atf-sdl-watchdog-service.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+#### Diagrams 
+![SDL watch dog and ATF communication](/assets/proposals/nnnn-Atf-Sdl-Watchdog-Service/sdl-watchdog.png)
+
+## Potential downsides
+
+N/A
+
+## Impact on existing code
+
+Impacts ATF code, some scripts
```
Comment (theresalech) [inline proposals/nnn-atf-sdl-watchdog-service.md]: Please provide more detail on the code and scripts impacted.

**Comment by LuxoftAKutsan** [comment]
@theresalech thanks,  fixed according to comments

File: proposals/nnn-atf-sdl-watchdog-service.md [hunk 5]
```diff
@@ -102,7 +102,8 @@ N/A
 
 ## Impact on existing code
 
-Impacts ATF code, some scripts
+Impacts atf component named SDL, also impacts Fasade of ATF. 
+Impacts all scripts that related to multiple ignition cycles
```
Comment (theresalech) [inline proposals/nnn-atf-sdl-watchdog-service.md]: Can you please update this to "Impacts all scripts related to multiple ignition cycles."?

Comment (LuxoftAKutsan) [reply proposals/nnn-atf-sdl-watchdog-service.md]: @theresalech done with  3f9ec02

**Pull request merged**